core/src/main/java/io/grpc/internal/SerializingExecutor.java

```diff
@@ -135,39 +135,31 @@ public void execute(Runnable r) {
   private class TaskRunner implements Runnable {
     @Override
     public void run() {
-      boolean stillRunning = true;
-      try {
-        while (true) {
-          Runnable nextToRun;
+      Runnable nextToRun;
+      ArrayDeque<Runnable> tmp = null;
+      do {
+        if (tmp == null) {
```

Comment:
This check seems strange, since `tmp` is guaranteed to be null at this point. I would have instead expected `if (consumerQueue.isEmpty()) {`